docs/proposals/error-handling-guideline.md

```diff
@@ -249,6 +250,42 @@ $ oras push localhost:5000/oras:v1 hello.txt
 Error: /home/user/hello.txt: no such file or directory
 ```
 
+#### Example 12: failed to authenticate with registry using an error credential from credential store
+
+Current behavior and output:
+
+```console
+$ oras pull localhost:7000/repo:tag --registry-config auth.config
+Error: failed to resolve tag: GET "http://localhost:7000/v2/repo/manifests/tag": credential required for basic auth
+```
+
+Suggested error message:
+
+```console
+$ oras pull localhost:7000/repo:tag --registry-config auth.config
+Error: pulling content from localhost:7000/repo:tag failed with status: 401 Unauthorized
+Please check whether the registry credential stored in the authentication file is correct
+```
+
+#### Example 13: failed to resolve the digest with incorrect username or password
+
+Current behavior and output:
+
+```console
+oras resolve localhost:7000/command/artifacts:foobar -u t -p 2
+WARNING! Using --password via the CLI is insecure. Use --password-stdin.
+Error response from registry: <nil>
+```
+
+Suggested error message:
+
+```console
+oras resolve localhost:7000/command/artifacts:foobar -u t -p 2
+WARNING! Using --password via the CLI is insecure. Use --password-stdin.
+Error: resolving the digest of artifact localhost:7000/command/artifacts:foobar failed with status: 401 Unauthorized
+Please use correct username or password
```

Comment:
We might want to rephase this sentence. Examples from Bing Chat:
1. "Invalid login credentials. Please try again."
2. "Authentication failed. Please verify your login details and try again."
3. "Access denied. Please ensure that your login information is correct and try again."

Reply:
Agree. *Login credentials* is much proper than *username/password* in this case. I updated this sentence to:

"Authentication failed. Please verify your login credentials and try again."